esp/bindings/http/platform/httpservice.cpp

```diff
@@ -417,6 +417,28 @@ int CEspHttpServer::processRequest()
             Owned<ISpan> serverSpan = m_request->createServerSpan(serviceName, methodName);
             ctx->setRequestSpan(serverSpan);
 
+            // Per https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/, the value
+            // of http.request.method must be "known to the instrumentation" and is case sensitive.
+            if (streq(method, POST_METHOD) || streq(method, GET_METHOD))
```

Comment:
The goal should be to annotate the active span w/ the actual http method type processed while minimizing new overhead.
This new logic relies on unnecessary string comparisons and explicitly tests for method types our server doesn't seem to support.
Why not annotate the span in-line within the actual post, get, unsupported code paths?

Reply:
I opted to keep the code together for illustrative purposes. ESP is not the only component that sets this attribute., although maybe it is the only component that needs to worry about anything other than POST.

I wanted to show what the generic expectation for the property is to setup a discussion about what we actually want to support. You and I already discussed offline a deviation from the standard, where an original lower or mixed case value is not recorded in addition to the canonical value..

Are we concerned about any values other than POST and GET? We weren't before this change, and maybe we still aren't.

If we want to record other methods, do we care about canonical versus original forms? Do we care about `_OTHER`?

Reply:
I personally don't know why we'd care about the original form (since our existing routing logic is case insensitive). I would focus on reflecting exactly how a request was processed.
>Are we concerned about any values other than POST and GET?
Only if we support any other values.
>If we want to record other methods, do we care about canonical versus original forms? Do we care about _OTHER?
Canonical form is preferred simply to adhere to the OTEL convention, but avoid/delay/minimize unneeded overhead.
I suppose _OTHER would only come in to play in an "unsupported" code path, so sure report _OTHER, and the original type.

Reply:
Would it be worthwhile to create a new Jira to define a method enumeration, map the text once in the request, and replace all of the string comparisons here (and anywhere else) with numerical comparisons? Finally add an inline method to map the enumeration to a canonical form to avoid case conversion on output.

This could be an alternative to using the string, not a replacement of it, so any uses of the string not in our control are unaffected.